app/models/invoice.rb

```diff
@@ -29,4 +29,8 @@ def total_revenue
     item_total = invoice_items.find_by_sql("select quantity * unit_price as item_total from invoice_items")
     item_total.pluck(:item_total).sum
   end
+
+  def self.incomplete
+    joins(:invoice_items).where.not(invoice_items: { status: 2 }).order(:created_at)
```

Comment:
Nice AR query!

Reply:
Seriously, after that first one, Joel is becoming the AR wizard